docs/messaging/messaging-metrics.md

```diff
@@ -0,0 +1,194 @@
+| `messaging.publish.messages` | Counter | `{message}` | Measures the number of published messages. |
+<!-- endsemconv -->
+
+## Consumer metrics
+
+### Metric: `messaging.receive.duration`
+
+This metric is [required][MetricRequired] when messaging system supports poll-based receive operations.
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.26.0/specification/metrics/api.md#instrument-advice)
+of `[ 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10 ]`.
+
+When this metric is reported alongside a messaging receive span, the metric value SHOULD be the same as the corresponding span duration.
+
+<!-- semconv metric.messaging.receive.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `messaging.receive.duration` | Histogram | `s` | Measures the duration of receive operation. |
+<!-- endsemconv -->
+
+### Metric: `messaging.receive.messages`
+
+This metric is [required][MetricRequired] when messaging system supports batch receive. It's [opt-in][MetricOptIn] when messaging system does not support batch receive since message count can be derived from `messaging.receive.duration` histogram.
+
+_Note: The need to report `messaging.receive.messages` depends on the messaging system capabilities and not application scenarios or client library limitations.
+
+<!-- semconv metric.messaging.receive.messages(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `messaging.receive.messages` | Counter | `{message}` | Measures the number of received messages. |
+<!-- endsemconv -->
+
+### Metric: `messaging.deliver.duration`
+
+This metric is [required][MetricRequired] when messaging system supports poll-based receive operations.
+
+When this metric is reported alongside a messaging deliver span, the metric value SHOULD be the same as the corresponding span duration.
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.26.0/specification/metrics/api.md#instrument-advice)
+of `[ 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10 ]`.
+
+<!-- semconv metric.messaging.deliver.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `messaging.deliver.duration` | Histogram | `s` | Measures the duration of deliver operation. |
+<!-- endsemconv -->
+
+### Metric: `messaging.deliver.messages`
+
+This metric is [required][MetricRequired] when messaging system supports batch delivery. It's [opt-in][MetricOptIn] when messaging system does not support batch delivery since message count can be derived from `messaging.deliver.duration` histogram.
```

Comment:
```suggestion
This metric is [required][MetricRequired] when the messaging system supports batch delivery. It's [opt-in][MetricOptIn] when the messaging system does not support batch delivery since the message count can be derived from the `messaging.deliver.duration` histogram.
```